src/common/libsubprocess/subprocess.h

```diff
@@ -184,6 +184,19 @@ flux_subprocess_t *flux_rexec_ex (flux_t *h,
                                   subprocess_log_f log_fn,
                                   void *log_data);
 
+/* Like flux_rexec(3), but run process in background.
+ *
+ * If service_name is NULL, then default to `rexec`.
+ *
+ * Once the process has been started, the returned future will be fulfilled
+ * with the payload ``{"rank":i, "pid":i}``, or an error response if there
+ * is an error.
+ */
```

Comment:
If we stick with the streaming RPC, it might be wise to mention that the rank/pid response will be immediately followed by ENODATA so people looking at the source don't see the STREAMING flag and wonder if other responses are possible.

Reply:
I'm fine with switching to a non-streaming RPC (in fact I think it is a good idea), if it is ok to have a single RPC that can operate as streaming or non-streaming depending on flags in the request.

Reply:
IMHO that should be OK.   In fact I wonder if the SUBPROCESS_REXEC_BACKGROUND flag would even be necessary then, since it seems like the logical way to handle the "non-streaming version" of the RPC.

Reply:
Yep, better!